packages/framework-provider-azure/src/helpers/query-helper.ts

```diff
@@ -214,3 +218,83 @@ function toLocalSortFor(
   })
   return sortedList
 }
+
+function buildProjections(projections: ProjectionFor<unknown> | string = '*'): string {
+  if (typeof projections !== 'object') {
+    return projections
+  }
+
+  // Group fields by the root property
+  const groupedFields: any = {}
+  Object.values(projections).forEach((field: string) => {
+    const root: string = field.split('.')[0]
+    if (!groupedFields[root]) {
+      groupedFields[root] = []
+    }
+    groupedFields[root].push(field)
+  })
+
+  return Object.keys(groupedFields)
+    .map((root: string): string => {
+      const fields = groupedFields[root]
+      if (root.endsWith('[]')) {
+        const arrayRoot: string = root.slice(0, -2)
+        const subFields = fields.map((f: string) => f.replace(`${root}.`, 'item.')).join(', ')
+        return `ARRAY(SELECT ${subFields} FROM item IN c.${arrayRoot}) AS ${arrayRoot}`
+      } else if (fields.length === 1 && !fields[0].includes('.')) {
+        // Simple field
+        return `c.${fields[0]}`
+      } else {
+        // Nested object fields
+        return fields.map((f: string) => `c.${f} AS "${f}"`).join(', ')
+      }
+    })
+    .join(', ')
+}
+
+function nestProperties(obj: any): any {
+  const result = {}
+
+  function setNestedProperty(obj: any, path: string[], value: any): void {
+    let current = obj
+    for (let i = 0; i < path.length - 1; i++) {
+      if (!current[path[i]]) {
+        current[path[i]] = {}
+      }
+      current = current[path[i]]
+    }
+    current[path[path.length - 1]] = value
+  }
+
+  function processObject(input: any, output: any): void {
+    for (const key in input) {
+      if (Object.prototype.hasOwnProperty.call(input, key)) {
+        const value = input[key]
+        const keys = key.split('.')
+        setNestedProperty(output, keys, value)
+      }
+    }
+  }
+
+  function processArray(arr: any[]): any[] {
+    return arr.map((item: any): any => {
+      if (Array.isArray(item)) {
+        return processArray(item)
```

Comment:
I can't grasp my head around this.. When we do a query to the DB, we either fetch one item (`{'a.b.c': 42}`) or an array of items (`[{'a.b.c': 42}, {'a.b.c': 43}, ...]`. Why/when would we get an array of arrays?

Reply:
@alvaroloes it shouldn't be necessary for the DB-specific case, as you correctly point out, but the `nestProperties` and its helper methods were made to be generic.

Reply:
All right